compiler/rustc_codegen_ssa/src/back/link.rs

```diff
@@ -385,8 +380,11 @@ fn link_rlib<'a>(
             tmpdir.as_ref(),
         );
 
-        ab.add_archive(&output_path, Box::new(|_| false)).unwrap_or_else(|e| {
-            sess.fatal(&format!("failed to add native library {}: {}", output_path.display(), e));
+        ab.add_archive(&output_path, Box::new(|_| false)).unwrap_or_else(|error| {
+            sess.emit_fatal(errors::AddNativeLibrary {
+                library_path: &output_path.display().to_string(),
```

Comment:
I'm pretty sure we support `Path` and `PathBuf` fields in the structs.

Reply:
Yes, but this error is being used in two places, and one of them uses `.to_string_lossy()` instead of PathBuf's Display implementation.

```rust
                sess.emit_fatal(errors::AddNativeLibrary {
                    library_path: &location.to_string_lossy(),
                    error,
                });
```

In this case, would you prefer to have two errors instead, or is it fine as is ?

Reply:
I would be surprised if the difference is important, you can probably just use `PathBuf`.

Reply:
Updated [here](https://github.com/rust-lang/rust/pull/102612/commits/954a62ff46cb8d6768e28a467e4393a780886576)